etl/steps/data/grapher/climate/2023-12-20/surface_temperature_annual_average.py

```diff
@@ -34,15 +37,34 @@ def run(dest_dir: str) -> None:
         )
         .reset_index()
     )
+
     # Convert the 'year' column to integer type
     tb_annual_average["year"] = tb_annual_average["year"].astype(int)
 
+    # Create a new column for the decade
+    tb_annual_average["decade"] = (tb_annual_average["year"] // 10) * 10
+
+    # Group by decade and country, then calculate the mean for specified columns
+    tb_decadal_average = (
+        tb_annual_average.groupby(["decade", "country"])[["temperature_anomaly", "temperature_2m"]].mean().reset_index()
+    )
+    # Set the decadal values for 2020 to NaN
+    tb_decadal_average.loc[tb_decadal_average["decade"] == 2020, ["temperature_anomaly", "temperature_2m"]] = np.nan
+    # Merge the decadal average Table with the original Table
+    combined = pr.merge(
+        tb_annual_average, tb_decadal_average, on=["decade", "country"], how="left", suffixes=("", "_decadal")
+    )
+
+    # Replace the decadal values with NaN for all years except the start of each decade
+    combined.loc[combined["year"] % 10 != 0, ["temperature_anomaly_decadal", "temperature_2m_decadal"]] = np.nan
+    combined = combined.drop(columns=["decade"])
     # Filter rows where the year is less than or equal to 2024
-    tb_annual_average = tb_annual_average[tb_annual_average["year"] < 2024]
-    tb_annual_average = tb_annual_average.set_index(["year", "country"])
+    combined = combined[combined["year"] < 2024]
```

Comment:
I'd suggest having 2024 as a parameter at the top of the script, or even better, calculated automatically as the latest year informed in the data.

Reply:
In general, I'd suggest to add `reset_index(drop=True)` every time you slice a table.